core/kazoo_number_manager/src/carriers/knm_carriers.erl

```diff
@@ -53,145 +53,149 @@ find(Num, Quantity) ->
+        {'bulk', []} -> Acc;
+        {'bulk', Numbers} -> process_bulk_carrier_results(Numbers, Acc);
+        {'stopping_here', Numbers} ->
+            NewAcc = process_carrier_results(Numbers, Acc),
+            NewAcc#{should_continue => 'false'};
+        {'error', _R} ->
+            lager:debug("skipping carrier ~s: ~p", [Carrier, _R]),
+            Acc
     catch
         _E:_R ->
             ST = erlang:get_stacktrace(),
             ?LOG_WARN("failed to query carrier ~s for ~p numbers: ~s: ~p"
-                      ,[Carrier, Quantity, _E, _R]
-                     ),
+                     ,[Carrier, Quantity, _E, _R]),
             log_stacktrace(ST),
             Acc
     end.
 
--spec process_bulk_carrier_results(kz_json:objects(), knm_number:knm_numbers()) ->
-                                          kz_json:objects().
-process_bulk_carrier_results(Acc, Numbers) ->
-    found_numbers_to_jobjs(Numbers) ++ Acc.
-
--spec process_carrier_results(kz_json:objects(), knm_number:knm_numbers()) ->
-                                     kz_json:objects().
-process_carrier_results(Acc, []) -> Acc;
-process_carrier_results(Acc, Numbers) ->
-    Results = lists:foldl(fun process_number_result/2, [], Numbers),
-    lists:reverse(Results, Acc).
+-spec process_bulk_carrier_results(knm_number:knm_numbers(), find_acc()) -> find_acc().
+process_bulk_carrier_results(Numbers, Acc) ->
+    acc_found(Acc, [found_number_to_jobj(Number) || Number <- Numbers]).
+
+-spec process_carrier_results(knm_number:knm_numbers(), find_acc()) -> find_acc().
+process_carrier_results(Numbers, Acc) ->
+    acc_found(Acc, lists:foldl(fun process_number_result/2, [], Numbers)).
+
+-spec acc_found(find_acc(), kz_json:objects()) -> find_acc().
+acc_found(Acc=#{found := Found
+               ,count := Count
+               ,left := Left
+               }, NewNumbers) ->
+    NewNumbersCount = length(NewNumbers),
+    Acc#{found => Found ++ NewNumbers
+        ,count => Count + NewNumbersCount
+        ,left => Left - NewNumbersCount
+        }.
 
 -spec process_number_result(knm_number:knm_number(), kz_json:objects()) ->
                                    kz_json:objects().
 process_number_result(Number, Acc) ->
     PhoneNumber = knm_number:phone_number(Number),
-    process_number_result(Number, Acc, knm_phone_number:module_name(PhoneNumber)).
-
-process_number_result(Number, Acc, ?CARRIER_OTHER) ->
-    [found_number_to_jobj(Number) | Acc];
-process_number_result(Number, Acc, Carrier) ->
-    DID = knm_phone_number:number(knm_number:phone_number(Number)),
-    check_for_existing_did(Number, Acc, Carrier, knm_phone_number:fetch(DID)).
+    case knm_phone_number:module_name(PhoneNumber) of
+        ?CARRIER_OTHER ->
+            [found_number_to_jobj(Number) | Acc];
+        Carrier ->
+            DID = knm_phone_number:number(PhoneNumber),
+            check_for_existing_did(Number, Acc, Carrier, knm_phone_number:fetch(DID))
+    end.
```

Comment:
why are you treating `?CARRIER_OTHER` differently ?
no carrier module should have an exception treatment, maybe you could extend the carrier behaviour for not having to deal with exceptions here.